src/FondOfSpryker/Zed/CollaborativeCart/Business/Releaser/CartReleaser.php

```diff
@@ -0,0 +1,114 @@
+        $this->quoteWriter = $quoteWriter;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ReleaseCartRequestTransfer $releaseCartRequestTransfer
+     *
+     * @return \Generated\Shared\Transfer\ReleaseCartResponseTransfer
+     */
+    public function release(ReleaseCartRequestTransfer $releaseCartRequestTransfer): ReleaseCartResponseTransfer
+    {
+        try {
+            $releaseCartResponseTransfer = $this->doRelease($releaseCartRequestTransfer);
+        } catch (Exception $exception) {
+            $releaseCartResponseTransfer = (new ReleaseCartResponseTransfer())
+                ->setIsSuccess(false)
+                ->setError($exception->getMessage());
+        }
+
+        return $releaseCartResponseTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ReleaseCartRequestTransfer $releaseCartRequestTransfer
+     *
+     * @return \Generated\Shared\Transfer\ReleaseCartResponseTransfer
+     */
+    protected function doRelease(ReleaseCartRequestTransfer $releaseCartRequestTransfer): ReleaseCartResponseTransfer
+    {
+        $quoteTransfer = $this->getQuoteByReleaseCartRequest($releaseCartRequestTransfer);
+
+        $companyUserReference = $quoteTransfer->getOriginalCompanyUserReference();
+        $customerReference = $quoteTransfer->getOriginalCustomerReference();
+
+        $quoteTransfer->setOriginalCompanyUserReference(null)
+            ->setOriginalCustomerReference(null)
+            ->setCompanyUserReference($companyUserReference)
+            ->setCustomerReference($customerReference)
+            ->setCustomer((new CustomerTransfer())->setCustomerReference($customerReference));
+
+        return (new ReleaseCartResponseTransfer())
+            ->setIsSuccess(true)
+            ->setQuote($this->updateQuote($quoteTransfer));
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ReleaseCartRequestTransfer $releaseCartRequestTransfer
+     *
+     * @throws \FondOfSpryker\Zed\CollaborativeCart\Business\Exception\QuoteCouldNotBeReleasedException
+     *
+     * @return \Generated\Shared\Transfer\QuoteTransfer
+     */
+    protected function getQuoteByReleaseCartRequest(ReleaseCartRequestTransfer $releaseCartRequestTransfer): QuoteTransfer
+    {
+        $quoteTransfer = $this->quoteReader->getByReleaseCartRequest($releaseCartRequestTransfer);
+
+        if ($quoteTransfer === null) {
+            throw new QuoteCouldNotBeReleasedException('Could not find quote to release.');
+        }
+
+        return $quoteTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @throws \FondOfSpryker\Zed\CollaborativeCart\Business\Exception\QuoteCouldNotBeReleasedException
+     *
+     * @return \Generated\Shared\Transfer\QuoteTransfer
+     */
+    protected function updateQuote(QuoteTransfer $quoteTransfer): QuoteTransfer
+    {
+        $quoteTransfer = $this->quoteWriter->update($quoteTransfer);
+
```

Comment:
Maybe if the error will appear in logs... It could be better to save the id or ref before update and add it to the exception